tests/lora/utils.py

```diff
@@ -1594,17 +1593,26 @@ def test_lora_fuse_nan(self):
                     ].weight += float("inf")
                 else:
                     named_modules = [name for name, _ in pipe.transformer.named_modules()]
-                    tower_name = (
-                        "transformer_blocks"
-                        if any(name == "transformer_blocks" for name in named_modules)
-                        else "blocks"
-                    )
-                    transformer_tower = getattr(pipe.transformer, tower_name)
-                    has_attn1 = any("attn1" in name for name in named_modules)
-                    if has_attn1:
-                        transformer_tower[0].attn1.to_q.lora_A["adapter-1"].weight += float("inf")
-                    else:
-                        transformer_tower[0].attn.to_q.lora_A["adapter-1"].weight += float("inf")
+                    possible_tower_names = [
+                        "transformer_blocks",
+                        "blocks",
+                        "joint_transformer_blocks",
+                        "single_transformer_blocks",
+                    ]
+                    filtered_tower_names = [
+                        tower_name for tower_name in possible_tower_names if hasattr(pipe.transformer, tower_name)
+                    ]
+                    if len(filtered_tower_names) == 0:
+                        pytest.xfail(
+                            reason=f"`pipe.transformer` didn't have any of the following attributes: {possible_tower_names}."
+                        )
```

Comment:
We can safely remove these. We don't really use `pytest.xfail` inside a test like this.

Reply:
Done in https://github.com/huggingface/diffusers/pull/10216/commits/d2ecef4b735d0d9b09e2067482dd223ec23e6c4b.